types/qs/qs-tests.ts

```diff
@@ -357,6 +357,12 @@ qs.parse("a=b&c=d", { delimiter: "&" });
     assert.deepEqual(sparseArray, { a: [, "2", , "5"] });
 });
 
+(() => {
+    var heterogeneousArray = qs.parse("user[0]=Alice&user[1][name]=Bob");
+    var expected: qs.ParsedQs = { user: [ 'Alice', { name: 'Bob' } ] };
```

Comment:
By declaring this variable outside of the `assert` call and explicitly declaring its type we get the run-time check of the parsing behavior as well as a type-time check on the value's shape.
This declaration resulted in a type error before my interface change, and compiles cleanly after the change.